### ISSUE TITLE
Use eval mode during validation

### ISSUE THREAD

**Issue by wutongshenqiu** [as791/ZOO_Attack_PyTorch#7]
Hi, thanks for the useful project written in pytorch. When looking into the code, I found that some places could be optimized:
1. Use `model.eval()` during [validation](https://github.com/as791/ZOO_Attack_PyTorch/blob/main/setup_mnist_model.py#L53). According to the documentation of pytorch, [`Dropout`](https://pytorch.org/docs/stable/generated/torch.nn.Dropout.html) behaves differently between 'train' and 'eval' mode. And both models used in cifar10 and mnist have `Dropout` layers.
2. Wrapping validation part with `with torch.no_grad()` may save time and memory.

**Comment by as791** [comment]
Hi can you please test the changes and raise PR for the same @wutongshenqiu

**Comment by webbrain-one** [comment]
Thanks for the great suggestions! 🎉

I've opened [a PR](https://github.com/as791/ZOO_Attack_PyTorch/pull/10) that addresses these points by adding `model.eval()` and wrapping the validation step with `torch.no_grad()`.

Thanks again for helping improve the project!
